webpack/assets/javascripts/react_app/components/PasswordStrength/__tests__/index.test.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { createStore, combineReducers } from 'redux';
+import { Provider } from 'react-redux';
+
+import PasswordStrength, { reducers } from '../index';
+
+const fixtures = {
+  username: { password: 'my-user-name', expected: __('Weak') },
+  email: { password: 'my-user@email.com', expected: __('Weak') },
+  tooShort: { password: '12345', expected: __('Too short') },
+  weak: { password: '123456', expected: __('Weak') },
+  medium: { password: 'qwedsa', expected: __('Medium') },
+  normal: { password: 'QwedsaZx', expected: __('Normal') },
+  strong: { password: 'StrongP@$w0rd', expected: __('Strong') },
+  veryStrong: { password: '|Kkh)Xeu#T8($"P;', expected: __('Very strong') },
+};
+
+document.getElementById = jest.fn(id => ({ value: fixtures[id].password }));
+
+describe('PasswordStrength integration test', () => {
+  const generateStore = () => createStore(combineReducers({ ...reducers }));
+
+  const mountComponent = store =>
+    mount(<Provider store={store}>
+        <div>
+          <input id="username" value={fixtures.username.password} readOnly />
+          <input id="email" value={fixtures.email.password} readOnly />
+          <PasswordStrength
+            data={{
+              className: 'form-control',
+              id: 'user_password',
+              name: 'user[password]',
+              verify: { name: 'user[password_confirmation]' },
+              userInputIds: ['username', 'email'],
+            }}
+          />
+        </div>
+      </Provider>);
+
+  const setInputValue = (input, value) => {
+    input.instance().value = value; // eslint-disable-line no-param-reassign
+    input.simulate('change', { target: { value } });
```

Comment:
I think you can do it in one line:
```javascript
input.simulate('change', {target: {value: 'My new value'}});
```

Reply:
Not in this case, i was disappointed to see that the plugin takes the value from the input instead of from the event.